cmd/dmverity-vhd/main.go

```diff
@@ -72,16 +74,26 @@ func main() {
 
 func fetchImageLayers(ctx *cli.Context) (layers []v1.Layer, err error) {
 	image := ctx.String(imageFlag)
+	tarballPath := ctx.String(tarballFlag)
 	ref, err := name.ParseReference(image)
 	if err != nil {
 		return nil, errors.Wrapf(err, "failed to parse image reference: %s", image)
 	}
 
-	local := ctx.GlobalBool(sourceFlag)
+	dockerDaemon := ctx.GlobalBool(dockerFlag)
 
 	// by default, using remote as source
 	var img v1.Image
-	if local {
+	if tarballPath != "" {
+		// create a tag and search the tarball for the image specified
+		var tag name.Tag
+		tag, err = name.NewTag(image)
```

Comment:
as discussed offline, can you document why we need to have a tag here? You mentioned that if no tag is passed, we'll try to read image with `:latest` tag, which is a reasonable thing to do IMO.

Reply:
Sorry, I didn't explain it very well. The output from `name.newTag()` is used to search for a particular image regardless of what its tag is in the sense of `image:tag`. `image` here is a required parameter so it'll always be non-empty, so the expected behavior assuming that is:

ex)
if image == "alpine" -> search for "alpine:latest"
and
if image == "alpine:3:16" -> search for "alpine:3.16"

in both cases, that `tag` variable is needed. I'll rename it to something like `imageNameAndTag` so it's clearer what's getting stored in there

Reply:
ah, ok. thanks for clarifying.